clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -498,8 +498,7 @@ class VariableAdderVisitor : public RecursiveASTVisitor<VariableAdderVisitor> {
 
   bool VisitVarDecl(VarDecl *D) {
     FullSourceLoc FL = Context->getFullLoc(D->getBeginLoc());
-    if (FL.isValid() &&
-        (!isa<ParmVarDecl>(D) || D->getParentFunctionOrMethod() != nullptr))
+    if (FL.isValid() && !isa<ParmVarDecl>(D))
```

Comment:
Was the `getParentFunctionOrMethod` redundant?

Reply:
I'm not sure what the original motivation for adding that clause to the conditional. `getParentFunctionOrMethod` returns true when the `ParmVarDecl` is associated with a function, but if it's associated with a function, it's added to the variable map at the same time as the function declaration. I thought maybe it was supposed to be `== nullptr` (i.e., it does not have an associated function), but this breaks lots of things.